app/Content/src/Http/Requests/BlogPosts/UpdateBlogPostRequest.php

```diff
@@ -0,0 +1,34 @@
+<?php
+
+namespace App\Content\Http\Requests\BlogPosts;
+
+use App\Platform\Enums\RoleName;
+use Illuminate\Foundation\Http\FormRequest;
+
+class UpdateBlogPostRequest extends FormRequest
+{
+    /**
+     * Determine if the user is authorized to make this request.
+     */
+    public function authorize(): bool
+    {
+        return $this->user()->hasRole(RoleName::ADMIN) || $this->user()->hasRole(RoleName::CONTENT_MANAGER);
+    }
+
+    /**
+     * Get the validation rules that apply to the request.
+     *
+     * @return array<string, \Illuminate\Contracts\Validation\ValidationRule|array<mixed>|string>
+     */
+    public function rules(): array
+    {
+        return [
+            'title' => ['sometimes'],
+            'excerpt' => ['sometimes'],
+            'slug' => ['sometimes'],
+            'content' => ['sometimes'],
+            'tags' => ['sometimes', 'array'],
+            'tags.*' => ['sometimes', 'exists:blog_post_tags,id']
```

Comment:
Same rules as above for all of these rules, but here, you will have to ignore the current tag name that is being edited in order to avoid the already exists error if the tag name has not changed.

Reply:
This may not work if you try to send the same data multiple times because the current tag IDs are not ignored currently. Please investigate this part.